Modules/_bisectmodule.c

```diff
@@ -39,6 +40,14 @@ internal_bisect_right(PyObject *list, PyObject *item, Py_ssize_t lo, Py_ssize_t
         litem = PySequence_GetItem(list, mid);
         if (litem == NULL)
             return -1;
+        if (key != Py_None) {
```

Comment:
Should this loop be duplicated for `key == Py_None` and `key != Py_None` like the Python implementation ? If I remember correctly it does not have much impact.

Reply:
In the C implementation, this test is essentially free because it is a 100% branch predicable pointer comparison.  In pure Python, every opcode in a loop adds cost.